src/envs/state_machine/main.cpp

```diff
@@ -68,6 +68,7 @@ int main()
         if (event.is_valid)
         {
             step_application(&app, (event_t)event.value);
+            step_state(&app.state_machines[app.active_state_machine_index], (event_t)event.value);
         }
         service_state_machine(&app.state_machines[app.active_state_machine_index]);
```

Comment:
I would make it like this:

```suggestion
        if (event.is_valid)
        {
            step_application(&app, (event_t)event.value);
        }
        service_application(&app);
```

The application should distribute the events and service the state machines.

Reply:
Resolved in 7bf5fc9